app/models/user.rb

```diff
@@ -34,17 +33,87 @@ class User < ApplicationRecord
          password_length: 10..128,
          otp_secret_encryption_key: ENV['ENCRYPTION_KEY'],
          otp_number_of_backup_codes: 10
-  # has_secure_password # not needed with devise
+
+  devise :omniauthable, omniauth_providers: [:okta] if ENV['OKTA_DOMAIN'].present?
+
+  def self.from_omniauth(auth)
+    user = find_by(
+      provider: auth['provider'],
+      uid: auth['uid'],
+    ) || find_by(
+      email: auth['info']['email'],
```

Comment:
We should confirm this is `downcase` coming out of OKTA, we've been bitten by case before.  I believe, we now correctly downcase all emails going into the DB.

Reply:
This is our token. created here https://github.com/greenriver/hmis-warehouse/blob/d6fb3378d7c67d1ffec4ed3e75e734dc4239c964/config/initializers/devise.rb#L412

The sloppy/annoying case conversions stuff is the `upcase` i do in the UI components. We should really have a translation there